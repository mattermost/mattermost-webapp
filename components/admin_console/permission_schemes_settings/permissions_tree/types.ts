// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type Permission = {
    id: string;
    combined?: boolean;
    permissions: string[];
}
export type Group = {
    id: string;
    permissions: Array<Permission | string>;
}

export type AdditionalValues = {
    edit_post: {
        editTimeLimitButton: JSX.Element;
    };
}
