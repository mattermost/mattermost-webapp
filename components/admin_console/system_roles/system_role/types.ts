// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type ReadAccess = 'read';
export const readAccess: ReadAccess = 'read';
export type WriteAccess = 'write';
export const writeAccess: WriteAccess = 'write';
export type NoAccess = false;
export const noAccess: NoAccess = false;
export type MixedAccess = 'mixed';
export const mixedAccess: MixedAccess = 'mixed';

export type PermissionAccess = ReadAccess | WriteAccess | NoAccess;
export type PermissionsToUpdate = Record<string, ReadAccess | WriteAccess | NoAccess>;
export type PermissionToUpdate = {
    name: string;
    value: PermissionAccess;
};

export type SystemSection = {
    name: string;
    hasDescription?: boolean;
    subsections?: SystemSection[];
    disabled?: boolean;
}
