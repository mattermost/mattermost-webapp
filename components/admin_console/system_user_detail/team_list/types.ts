// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Team, TeamMembership} from '@mattermost/types/teams';

export type TeamWithMembership = Team & TeamMembership;
