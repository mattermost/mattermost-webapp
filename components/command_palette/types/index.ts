// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export enum CommandPaletteEntities {
    Channel = 'channel',
    Playbooks = 'playbooks',
    Boards = 'boards',
    Messages = 'messages',
    Files = 'files',
    Runs = 'runs',
    GoTo = 'goTo',
}

export enum ChannelType {
    DM_CHANNEL= 'D',
    GM_CHANNEL= 'G',
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

export enum BoardsType {
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

export enum PlaybooksType {
    OPEN_CHANNEL= 'O',
    PRIVATE_CHANNEL= 'P',
}

