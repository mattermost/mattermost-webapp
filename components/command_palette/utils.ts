// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {Channel} from '@mattermost/types/channels';
import {Team} from '@mattermost/types/teams';

// See LICENSE.txt for license information.
import {CommandPaletteItem} from './command_palette_list_item/command_palette_list_item';
import {
    CmdPalettePictographIcon,
    CmdPalettePictographType,
} from './command_palette_list_item/command_palette_list_item_pictograph';
import {ChannelType, CommandPaletteEntities} from './types';

const channelTypeOrder = {
    [ChannelType.OPEN_CHANNEL]: 0,
    [ChannelType.PRIVATE_CHANNEL]: 1,
    [ChannelType.DM_CHANNEL]: 2,
    [ChannelType.GM_CHANNEL]: 3,
};

function filterName(name: string): string {
    return name.replace(/[.,'"\/#!$%\^&\*;:{}=\-_`~()]/g, ''); // eslint-disable-line no-useless-escape
}

export function channelToCommandPaletteItemTransformer(channels: Channel[], teams: Record<string, Team>): CommandPaletteItem[] {
    return channels.map((channel) => {
        const isArchived = channel.delete_at ? channel.delete_at !== 0 : false;

        let channelIcon = CmdPalettePictographIcon.GLOBE;
        if (isArchived) {
            channelIcon = CmdPalettePictographIcon.ARCHIVE;
        }
        return {
            description: channel.type === ChannelType.DM_CHANNEL ? channel.name : '',
            id: channel.id,
            isDeactivated: isArchived,
            teamName: teams[channel.team_id]?.name,
            pictograph: {
                type: CmdPalettePictographType.ICON,
                pictographItem: channelIcon,
            },
            subType: channel.type as ChannelType,
            title: channel.display_name,
            type: CommandPaletteEntities.Channel,
            isArchived,
        };
    });
}

function sortChannelsByTypeAndDisplayName(locale: string, cmdPaletteItemA: CommandPaletteItem, cmdPaletteItemB: CommandPaletteItem): number {
    if (channelTypeOrder[cmdPaletteItemA.subType] !== channelTypeOrder[cmdPaletteItemB.subType]) {
        if (channelTypeOrder[cmdPaletteItemA.subType] < channelTypeOrder[cmdPaletteItemB.subType]) {
            return -1;
        }

        return 1;
    }

    const aDisplayName = filterName(cmdPaletteItemA.title);
    const bDisplayName = filterName(cmdPaletteItemB.title);

    if (aDisplayName !== bDisplayName) {
        return aDisplayName.toLowerCase().localeCompare(bDisplayName.toLowerCase(), locale, {numeric: true});
    }

    return cmdPaletteItemA.title.toLowerCase().localeCompare(cmdPaletteItemA.title.toLowerCase(), locale, {numeric: true});
}

function sortChannelsByRecencyAndTypeAndDisplayName(cmdPaletteItemA: CommandPaletteItem, cmdPaletteItemB: CommandPaletteItem) {
    if (cmdPaletteItemA.lastViewedAt && cmdPaletteItemB.lastViewedAt) {
        return cmdPaletteItemB.lastViewedAt - cmdPaletteItemA.lastViewedAt;
    } else if (cmdPaletteItemA.lastViewedAt) {
        return -1;
    } else if (cmdPaletteItemB.lastViewedAt) {
        return 1;
    }

    // MM-12677 When this is migrated this needs to be fixed to pull the user's locale
    return sortChannelsByTypeAndDisplayName('en', cmdPaletteItemA, cmdPaletteItemB);
}

export function commandPaletteSorter(prefix: string, cmdPaletteItemA: CommandPaletteItem, cmdPaletteItemB: CommandPaletteItem) {
    if (cmdPaletteItemA.isArchived && !cmdPaletteItemB.isArchived) {
        return 1;
    } else if (!cmdPaletteItemA.isArchived && cmdPaletteItemB.isArchived) {
        return -1;
    }

    if (cmdPaletteItemA.isDeactivated && !cmdPaletteItemB.isDeactivated) {
        return 1;
    } else if (cmdPaletteItemB.isDeactivated && !cmdPaletteItemA.isDeactivated) {
        return -1;
    }

    let aDisplayName = cmdPaletteItemA.title.toLowerCase();
    let bDisplayName = cmdPaletteItemA.title.toLowerCase();

    if (cmdPaletteItemA.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemA.subType === ChannelType.DM_CHANNEL &&
        aDisplayName.startsWith('@')) {
        aDisplayName = aDisplayName.substring(1);
    }

    if (cmdPaletteItemB.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemB.subType === ChannelType.DM_CHANNEL && bDisplayName.startsWith('@')) {
        bDisplayName = bDisplayName.substring(1);
    }

    const aStartsWith = aDisplayName.startsWith(prefix) || cmdPaletteItemA.description.toLowerCase().startsWith(prefix);
    const bStartsWith = bDisplayName.startsWith(prefix) || cmdPaletteItemB.description.toLowerCase().startsWith(prefix);

    // Open channels user haven't interacted should be at the  bottom of the list
    if (cmdPaletteItemA.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemA.subType === ChannelType.OPEN_CHANNEL &&
        !cmdPaletteItemA.lastViewedAt &&
        ((cmdPaletteItemB.type === CommandPaletteEntities.Channel &&
            cmdPaletteItemB.subType !== ChannelType.OPEN_CHANNEL) || cmdPaletteItemB.lastViewedAt)) {
        return 1;
    } else if ((cmdPaletteItemB.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemB.subType === ChannelType.OPEN_CHANNEL) || !cmdPaletteItemB.lastViewedAt) {
        return -1;
    }

    // Sort channels starting with the search term first
    if (aStartsWith && !bStartsWith) {
        return -1;
    } else if (!aStartsWith && bStartsWith) {
        return 1;
    }
    return sortChannelsByRecencyAndTypeAndDisplayName(cmdPaletteItemA, cmdPaletteItemB);
}
