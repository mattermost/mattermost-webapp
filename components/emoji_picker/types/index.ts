// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {EmojiCategory, Emoji} from 'mattermost-redux/types/emojis';

export type Category = {
    className: string;
    emojiIds?: string[];
    id: string;
    message: string;
    name: EmojiCategory;
    offset: number;
};

export type Categories = Record<EmojiCategory, Category>;

export type CategoryOrEmojiRow = {
    type: 'categoryHeaderRow' | 'emojisRow';
    row: EmojiCategory | Emoji[];
}
