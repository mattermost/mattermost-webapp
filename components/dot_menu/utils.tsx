// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {trackEvent} from 'actions/telemetry_actions';

import {EventTypes, TELEMETRY_CATEGORIES} from 'utils/constants';

export type ChangeEvent = React.KeyboardEvent | React.MouseEvent;

export const trackDotMenuEvent = (e: ChangeEvent, suffix: string): void => {
    if (e.type === EventTypes.CLICK) {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.CLICK + '_' + suffix);
    } else {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.SHORTCUT + '_ ' + suffix);
    }
};
