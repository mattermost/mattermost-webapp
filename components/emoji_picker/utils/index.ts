// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Emoji, EmojiCategory} from 'mattermost-redux/types/emojis';

import {compareEmojis} from 'utils/emoji_utils';

import {Categories} from '../types';

function sortEmojis(emojis: Emoji[], recentEmojiStrings: string[], filter: string) {
    const recentEmojis: Emoji[] = [];
    const emojisMinusRecent: Emoji[] = [];

    Object.values(emojis).forEach((emoji) => {
        let emojiArray = emojisMinusRecent;
        const alias = 'short_names' in emoji ? emoji.short_names : [emoji.name];
        for (let i = 0; i < alias.length; i++) {
            if (recentEmojiStrings.includes(alias[i].toLowerCase())) {
                emojiArray = recentEmojis;
            }
        }

        emojiArray.push(emoji);
    });

    return [
        ...recentEmojis.sort((firstEmoji, secondEmoji) => compareEmojis(firstEmoji, secondEmoji, filter)),
        ...emojisMinusRecent.sort((firstEmoji, secondEmoji) => compareEmojis(firstEmoji, secondEmoji, filter)),
    ];
}

function getFilteredEmojis(allEmojis: Record<string, Emoji>, filter: string, recentEmojisString: string[]): Emoji[] {
    const emojis = Object.values(allEmojis).filter((emoji) => {
        const alias = 'short_names' in emoji ? emoji.short_names : [emoji.name];
        for (let i = 0; i < alias.length; i++) {
            if (alias[i].toLowerCase().includes(filter)) {
                return true;
            }
        }

        return false;
    });

    return sortEmojis(emojis, recentEmojisString, filter);
}

export function getEmojisByCategory(allEmojis: Record<string, Emoji>, categories: Categories, categoryName: EmojiCategory, filter: string, recentEmojisString: string[]): Emoji[] {
    if (filter.length) {
        return getFilteredEmojis(allEmojis, filter, recentEmojisString);
    }

    // if (categoryName === 'recent') {
    //     const recentEmojiIds = this.state.categories?.recent?.emojiIds ?? [];
    //     if (recentEmojiIds.length === 0) {
    //         return [];
    //     }

    //     const recentEmojis = new Map();

    //     recentEmojiIds.forEach((emojiId) => {
    //         const emoji = this.state.allEmojis[emojiId];
    //         const emojiSkin = getSkin(emoji);

    //         if (emojiSkin) {
    //             const emojiWithUserSkin = this.convertEmojiToUserSkinTone(emoji, emojiSkin, this.props.userSkinTone);
    //             recentEmojis.set(emojiWithUserSkin.unified, emojiWithUserSkin);
    //         } else {
    //             recentEmojis.set(emojiId, emoji);
    //         }
    //     });

    //     return Array.from(recentEmojis.values());
    // }

    return categories[categoryName].emojiIds.map((emojiId) => allEmojis[emojiId]);
}
