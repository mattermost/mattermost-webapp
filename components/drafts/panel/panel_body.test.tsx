// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {Provider} from 'react-redux';

import mockStore from 'tests/test_store';

import {PostDraft} from 'types/store/draft';

import * as utils from 'utils/utils';

import {mountWithIntl} from 'tests/helpers/intl-test-helper';

import {UserProfile, UserStatus} from '@mattermost/types/users';
import {PostPriority} from '@mattermost/types/posts';

import PanelBody from './panel_body';

describe('components/drafts/panel/panel_body', () => {
    const baseProps = {
        channelId: 'channel_id',
        displayName: 'display_name',
        fileInfos: [] as PostDraft['fileInfos'],
        message: 'message',
        status: 'status' as UserStatus['status'],
        uploadsInProgress: [] as PostDraft['uploadsInProgress'],
        userId: 'user_id' as UserProfile['id'],
        username: 'username' as UserProfile['username'],
    };

    const initialState = {
        entities: {
            general: {
                config: {},
            },
            posts: {
                posts: {
                    root_id: {id: 'root_id', channel_id: 'channel_id'},
                },
            },
            channels: {
                currentChannelId: 'channel_id',
                channels: {
                    channel_id: {id: 'channel_id', team_id: 'team_id'},
                },
            },
            preferences: {
                myPreferences: {},
            },
            groups: {
                groups: {},
                myGroups: [],
            },
            emojis: {
                customEmoji: {},
            },
            users: {
                currentUserId: 'userid1',
                profiles: {userid1: {id: 'userid1', username: 'username1', roles: 'system_user'}},
                profilesInChannel: {},
            },
            teams: {
                currentTeamId: 'team_id',
                teams: {
                    team_id: {
                        id: 'team_id',
                        name: 'team-id',
                        display_name: 'Team ID',
                    },
                },
            },
        },
    };

    it('should match snapshot', () => {
        const store = mockStore(initialState);

        const wrapper = mountWithIntl(
            <Provider store={store}>
                <PanelBody
                    {...baseProps}
                />
            </Provider>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should match snapshot for requested_ack', () => {
        const store = mockStore(initialState);

        const wrapper = mountWithIntl(
            <Provider store={store}>
                <PanelBody
                    {...baseProps}
                    priority={{
                        priority: '',
                        requested_ack: true,
                    }}
                />
            </Provider>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should match snapshot for priority', () => {
        const store = mockStore(initialState);

        const wrapper = mountWithIntl(
            <Provider store={store}>
                <PanelBody
                    {...baseProps}
                    priority={{
                        priority: PostPriority.IMPORTANT,
                        requested_ack: true,
                    }}
                />
            </Provider>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should have called handleFormattedTextClick', () => {
        const handleClickSpy = jest.spyOn(utils, 'handleFormattedTextClick');
        const store = mockStore(initialState);

        const wrapper = mountWithIntl(
            <Provider store={store}>
                <PanelBody
                    {...baseProps}
                />
            </Provider>,
        );

        wrapper.find('div.post__content').simulate('click');
        expect(handleClickSpy).toHaveBeenCalledTimes(1);
        expect(wrapper).toMatchSnapshot();
    });
});
