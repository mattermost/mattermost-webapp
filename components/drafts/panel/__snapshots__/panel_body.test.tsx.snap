// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/drafts/panel/panel_body should have called handleFormattedTextClick 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <PanelBody
    channelId="channel_id"
    displayName="display_name"
    fileInfos={Array []}
    message="message"
    status="status"
    uploadsInProgress={Array []}
    userId="user_id"
    username="username"
  >
    <div
      className="DraftPanelBody post"
    >
      <div
        className="DraftPanelBody__left post__img"
      >
        <ProfilePicture
          channelId="channel_id"
          hasMention={false}
          isEmoji={false}
          isRHS={false}
          popoverPlacement="right"
          size="md"
          src="/api/v4/users/user_id/image?_=0"
          status="status"
          userId="user_id"
          username="username"
          wrapperClass=""
        >
          <OverlayTrigger
            defaultOverlayShown={false}
            overlay={
              <Memo(Connect(injectIntl(ProfilePopover)))
                channelId="channel_id"
                className="user-profile-popover"
                hasMention={false}
                hide={[Function]}
                isRHS={false}
                src="/api/v4/users/user_id/image?_=0"
                userId="user_id"
              />
            }
            placement="right"
            rootClose={true}
            trigger={
              Array [
                "click",
              ]
            }
          >
            <OverlayTrigger
              defaultOverlayShown={false}
              overlay={
                <OverlayWrapper
                  channelId="channel_id"
                  className="user-profile-popover"
                  hasMention={false}
                  hide={[Function]}
                  intl={
                    Object {
                      "defaultFormats": Object {},
                      "defaultLocale": "en",
                      "defaultRichTextElements": undefined,
                      "formatDate": [Function],
                      "formatDateTimeRange": [Function],
                      "formatDateToParts": [Function],
                      "formatDisplayName": [Function],
                      "formatList": [Function],
                      "formatListToParts": [Function],
                      "formatMessage": [Function],
                      "formatNumber": [Function],
                      "formatNumberToParts": [Function],
                      "formatPlural": [Function],
                      "formatRelativeTime": [Function],
                      "formatTime": [Function],
                      "formatTimeToParts": [Function],
                      "formats": Object {},
                      "formatters": Object {
                        "getDateTimeFormat": [Function],
                        "getDisplayNames": [Function],
                        "getListFormat": [Function],
                        "getMessageFormat": [Function],
                        "getNumberFormat": [Function],
                        "getPluralRules": [Function],
                        "getRelativeTimeFormat": [Function],
                      },
                      "locale": "en",
                      "messages": Object {},
                      "onError": [Function],
                      "textComponent": "span",
                      "timeZone": "Etc/UTC",
                      "wrapRichTextChunksInFragment": undefined,
                    }
                  }
                  isRHS={false}
                  src="/api/v4/users/user_id/image?_=0"
                  userId="user_id"
                />
              }
              placement="right"
              rootClose={true}
              trigger={
                Array [
                  "click",
                ]
              }
            >
              <button
                className="status-wrapper style--none "
                onClick={[Function]}
                tabIndex={-1}
              >
                <span
                  className="profile-icon "
                >
                  <Memo(Avatar)
                    size="md"
                    url="/api/v4/users/user_id/image?_=0"
                    username="username"
                  >
                    <img
                      alt="username profile image"
                      className="Avatar Avatar-md"
                      onError={[Function]}
                      src="/api/v4/users/user_id/image?_=0"
                      tabIndex={0}
                    />
                  </Memo(Avatar)>
                </span>
                <StatusIcon
                  button={false}
                  className=""
                  status="status"
                >
                  <StatusOfflineIcon
                    className="status "
                  >
                    <span
                      className="status "
                    >
                      <svg
                        aria-label="Offline Icon"
                        className="offline--icon"
                        height="100%"
                        role="img"
                        style={
                          Object {
                            "clipRule": "evenodd",
                            "fillRule": "evenodd",
                            "strokeLinejoin": "round",
                            "strokeMiterlimit": 1.41421,
                          }
                        }
                        viewBox="0 0 20 20"
                        width="100%"
                      >
                        <path
                          d="M10,0c5.519,0 10,4.481 10,10c0,5.519 -4.481,10 -10,10c-5.519,0 -10,-4.481 -10,-10c0,-5.519 4.481,-10 10,-10Zm0,2c4.415,0 8,3.585 8,8c0,4.415 -3.585,8 -8,8c-4.415,0 -8,-3.585 -8,-8c0,-4.415 3.585,-8 8,-8Z"
                        />
                      </svg>
                    </span>
                  </StatusOfflineIcon>
                </StatusIcon>
              </button>
            </OverlayTrigger>
          </OverlayTrigger>
        </ProfilePicture>
      </div>
      <div
        className="post__content"
        onClick={[Function]}
      >
        <div
          className="DraftPanelBody__right"
        >
          <div
            className="post__header"
          >
            <strong>
              display_name
            </strong>
          </div>
          <div
            className="post__body"
          >
            <Connect(Markdown)
              message="message"
              options={
                Object {
                  "disableGroupHighlight": true,
                  "mentionHighlight": false,
                }
              }
            >
              <Markdown
                autolinkedUrlSchemes={
                  Array [
                    "http",
                    "https",
                    "ftp",
                    "mailto",
                    "tel",
                    "mattermost",
                  ]
                }
                channelNamesMap={Object {}}
                dispatch={[Function]}
                editedAt={0}
                emojiMap={
                  EmojiMap {
                    "customEmojis": Map {},
                    "customEmojisArray": Array [],
                  }
                }
                enableFormatting={true}
                hasImageProxy={false}
                imagesMetadata={Object {}}
                isRHS={false}
                managedResourcePaths={Array []}
                mentionKeys={Array []}
                message="message"
                minimumHashtagLength={NaN}
                options={
                  Object {
                    "disableGroupHighlight": true,
                    "mentionHighlight": false,
                  }
                }
                postId=""
                proxyImages={true}
                siteURL="http://localhost:8065"
                team={
                  Object {
                    "display_name": "Team ID",
                    "id": "team_id",
                    "name": "team-id",
                  }
                }
              >
                <p
                  key="0"
                >
                  message
                </p>
              </Markdown>
            </Connect(Markdown)>
          </div>
        </div>
      </div>
    </div>
  </PanelBody>
</Provider>
`;

exports[`components/drafts/panel/panel_body should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <PanelBody
    channelId="channel_id"
    displayName="display_name"
    fileInfos={Array []}
    message="message"
    status="status"
    uploadsInProgress={Array []}
    userId="user_id"
    username="username"
  />
</ContextProvider>
`;
